src/system-update-generator/system-update-generator.c

```diff
@@ -18,19 +19,21 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
+        _cleanup_free_ char *j = NULL;
+        FOREACH_STRING(p, "/system-update", "/etc/system-update") {
+                if (laccess(p, F_OK) >= 0)
+                        goto link_found;
 
-        if (laccess("/system-update", F_OK) < 0) {
-                if (errno == ENOENT)
-                        return 0;
-
-                log_error_errno(errno, "Failed to check for system update: %m");
-                return -EINVAL;
+                if (errno != ENOENT)
+                        log_error_errno(errno, "Failed to check for system update: %m");
```

Comment:
The failure will be ignored. Please downgrade the log level, and append `", ignoring"`.

Reply:
Added ", ignoring" plus I reworded the error message to actually include the symlink it failed the check for...